docs/js-api.md

```diff
@@ -0,0 +1,168 @@
+# Using Spectral programatically
+
+## Example: Linting an OpenAPI document
+
+Spectral includes a number of ready made rules and functions for OpenAPI v2 and v3 documents.
+
+This example uses the OpenAPI v3 rules to lint a document.
+
+```js
+const { Spectral } = require('@stoplight/spectral');
+const { oas3Functions, rules: oas3Rules } = require('@stoplight/spectral/rulesets/oas3');
+// for YAML
+const { parseWithPointers } = require("@stoplight/yaml");
+
+// Using the anonymous async wrapper for the sake of the example.
+const myOAS = parseWithPointers(`
+responses:
+  '200':
+    description: ''
+    schema:
+      $ref: '#/definitions/error-response'
+`)
+
+// an OASv3 document
+const myOAS = {
```

Comment:
```➜  Stoplight node test-spectral.js
/Users/taylorbarnett/Stoplight/test-spectral.js:16
const myOAS = {
      ^

SyntaxError: Identifier 'myOAS' has already been declared
    at new Script (vm.js:80:7)
    at createScript (vm.js:274:10)
    at Object.runInThisContext (vm.js:326:10)
    at Module._compile (internal/modules/cjs/loader.js:664:28)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:712:10)
    at Module.load (internal/modules/cjs/loader.js:600:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:539:12)
    at Function.Module._load (internal/modules/cjs/loader.js:531:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:754:12)
    at startup (internal/bootstrap/node.js:283:19)```

😉 make sure you run these?

Reply:
Hey @tbarn I think one is meant to comment out the myOAS they don't need (see that there are two ways we create myOas const). I did run it btw and just commented out one of the consts. Actually I will comment that out in the example so that one can copy paste and run.